core/src/main/java/i5/las2peer/registry/ReadWriteRegistryClient.java

```diff
@@ -304,12 +305,24 @@ public void registerService(String serviceName, EthereumAgent agent) throws Ethe
 		String consentee = agent.getEthereumAddress();
 		byte[] signature = SignatureUtils.signFunctionCall(function, agent.getEthereumCredentials());
 
+		if ( txMan != null )
+		{
+			BigInteger txManNonce = txMan.getCurrentNonce();
+			logger.info("[TX Nonce] before: " + txManNonce);
+		}
+
 		try {
 			contracts.serviceRegistry.delegatedRegister(serviceName, authorName, consentee, signature).sendAsync()
 			.get();
 		} catch (Exception e) {
 			throw new EthereumException("Failed to register service", e);
 		}
+
+		if ( txMan != null )
+		{
+			BigInteger txManNonce = txMan.getCurrentNonce();
+			logger.info("[TX Nonce] after: " + txManNonce);
+		}
 	}
```

Comment:
Not sure if the duplicate here is intended or not...

Reply:
I would say it is intended and helps verifying whether the nonce really increased.